src/irr/asset/CGLTFLoader.cpp

```diff
@@ -0,0 +1,1224 @@
+			{
+				if (key == "asset")
+				{
+					tweets.at_key("asset").at_key("version").get(element);
+					header.version = std::stoi(element.get_string().value().data());
+
+					auto& minVersion = value.at_key("minVersion");
+					if (minVersion.error() != simdjson::error_code::NO_SUCH_FIELD)
+					{
+						header.minVersion = minVersion.get_uint64().value();
+						if (header.minVersion.value() > header.version)
+							return {};
+					}
+
+					auto& generator = value.at_key("generator");
+					if (generator.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.generator = generator.get_string().value().data();
+
+					auto& copyright = value.at_key("copyright");
+					if (copyright.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.copyright = copyright.get_string().value().data();
+				}
+
+				
+					Buffers and buffer views do not contain type information.
+					They simply define the raw data for retrieval from the file.
+					Objects within the glTF file (meshes, skins, animations) access buffers
+					or buffer views via accessors.
+				
+
+				else if (key == "buffers")
+				{
+					for (auto& buffer : value)
+					{
+						auto& byteLength = buffer.at_key("byteLength");
+						if (byteLength.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							auto byteLengthVal = byteLength.get_uint64().value();
+							if (byteLengthVal < 1)
+								continue;
+						}
+						else
+							continue;
+
+						auto& uri = buffer.at_key("uri");
+						if (uri.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							std::string_view uriBin = uri.get_string().value();
+
+							const asset::IAssetLoader::SAssetLoadParams params;
+							auto buffer_bundle = assetManager->getAsset(rootAssetDirectory + uriBin.data(), params);
+							auto buffer = core::smart_refctd_ptr_static_cast<ICPUBuffer>(buffer_bundle.getContents().begin()[0]);
+
+							// put
+						}
+						else
+							continue;
+					}
+				}
+
+				else if (key == "bufferViews")
+				{
+					for (auto& bufferView : value)
+					{
+						asset::SBufferBinding<ICPUBuffer> bufferBinding;
+
+						auto& buffer = bufferView.at_key("buffer");
+						if (buffer.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							auto& bufferID = buffer.get_uint64().value();
+						}
+						else
+							continue;
+					}
+				}
+
+				
+					 Meshes are defined as arrays of primitives.
+					 Primitives correspond to the data required for GPU draw calls
+				
+
+				else if (key == "meshes")
+				{
+					for (auto& mesh : value)
+					{
+						auto& primitives = mesh.at_key("primitives");
+						if (primitives.error() != simdjson::error_code::NO_SUCH_FIELD)
+							for (auto& primitive : primitives)
+							{
+								auto& attributes = primitive.at_key("attributes");
+								if (attributes.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& position = attributes.at_key("POSITION");
+									auto& normal = attributes.at_key("NORMAL");
```

Comment:
ignore this attribute, we'll always use pixel-shader tangent construction.

Reply:
solved, not absent anymore